File: _CodeTopics/LeetCode_contest/biweekly/biweekly2020/26/26_1.py
class Solution(object):
    def maxPower(self, s):
        """
        :type s: str
        :rtype: int
        """
        
        maxenergy = 1
        length = len(s)
        for i in range(length):
            currenergy = 1
            for j in range(i+1, length):
                if s[j] == s[i]:
                    currenergy += 1
                    if currenergy > maxenergy:
                        maxenergy = currenergy
                else:
                    break
        return maxenergy
    

File: _CodeTopics/LeetCode/1401-1600/001446--biweekly26_1/tran/26_1.cpp
class Solution {
public:
    int maxPower(string s) {

        // 第 26 场双周赛第一题
        // https://leetcode-cn.com/submissions/detail/71312850/

        int maxenergy = 1;
        int length = s.size();
        for (int i = 0; i < length - 1; ++i) {
            int currenergy = 1;
            for (int j = i + 1; j < length; ++j) {
                if (s[j] == s[i]) {
                    currenergy += 1;
                    maxenergy = max(maxenergy, currenergy);
                } else {
                    break;
                }
            }
        }
        return maxenergy;
    }
};

/*
https://leetcode-cn.com/submissions/detail/244008655/

执行用时：0 ms, 在所有 C++ 提交中击败了100.00%的用户
内存消耗：6.7 MB, 在所有 C++ 提交中击败了22.74%的用户
通过测试用例：
333 / 333
*/
